tests/bedrock/test_bedrock_autolog.py

```diff
@@ -0,0 +1,358 @@
+_META_LLAMA_REQUEST = {
+    "prompt": """
+<|begin_of_text|><|start_header_id|>user<|end_header_id|>
+Hi!
+<|eot_id|>
+<|start_header_id|>assistant<|end_header_id|>
+""",
+    "max_gen_len": 300,
+    "temperature": 0.1,
+}
+
+_META_LLAMA_RESPONSE = {
+    "generation": "Hello! How can I help you today?",
+    "prompt_token_count": 2,
+    "generation_token_count": 12,
+    "stop_reason": "stop",
+}
+
+
+def _create_dummy_invoke_model_response(llm_response):
+    llm_response_encoded = json.dumps(llm_response).encode("utf-8")
+    return {
+        "body": StreamingBody(io.BytesIO(llm_response_encoded), len(llm_response_encoded)),
+        "ResponseMetadata": {
+            "RequestId": "request-id-123",
+            "HTTPStatusCode": 200,
+            "HTTPHeaders": {"content-type": "application/json"},
+            "RetryAttempts": 0,
+        },
+        "contentType": "application/json",
+    }
+
+
+@pytest.mark.parametrize(
+    ("model_id", "llm_request", "llm_response"),
+    [
+        (
+            "anthropic.claude-3-5-sonnet-20241022-v2:0",
+            _ANTHROPIC_REQUEST,
+            _ANTHROPIC_RESPONSE,
+        ),
+        (
+            "ai21.jamba-instruct-v1:0",
+            _AI21_JAMBA_REQUEST,
+            _AI21_JAMBA_RESPONSE,
+        ),
+        (
+            "us.amazon.nova-lite-v1:0",
+            _AMAZON_NOVA_REQUEST,
+            _AMAZON_NOVA_RESPONSE,
+        ),
+        (
+            "cohere.command-r-plus-v1:0",
+            _COHERE_REQUEST,
+            _COHERE_RESPONSE,
+        ),
+        (
+            "meta.llama3-8b-instruct-v1:0",
+            _META_LLAMA_REQUEST,
+            _META_LLAMA_RESPONSE,
+        ),
+    ],
+)
+def test_bedrock_autolog_invoke_model_llm(model_id, llm_request, llm_response):
+    mlflow.bedrock.autolog()
+
+    client = boto3.client("bedrock-runtime", region_name="us-west-2")
+    request_body = json.dumps(llm_request)
+
+    # Ref: https://docs.getmoto.org/en/latest/docs/services/patching_other_services.html
+    with mock.patch(
+        "botocore.client.BaseClient._make_api_call",
+        return_value=_create_dummy_invoke_model_response(llm_response),
+    ):
+        response = client.invoke_model(body=request_body, modelId=model_id)
+
+    response_body = json.loads(response["body"].read())
+    assert response_body == llm_response
+
+    traces = get_traces()
+    assert len(traces) == 1
+    assert traces[0].info.status == "OK"
+
+    assert len(traces[0].data.spans) == 1
+    span = traces[0].data.spans[0]
+    assert span.name == "BedrockRuntime.invoke_model"
+    assert span.span_type == "LLM"
+    assert span.inputs == {"body": request_body, "modelId": model_id}
+    assert span.outputs == {
+        "body": llm_response,
+        "ResponseMetadata": response["ResponseMetadata"],
+        "contentType": "application/json",
+    }
+
+
+# https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-titan-embed-text.html
+_AMAZON_EMBEDDING_REQUEST = {
```

Comment:
nit: shall we define these consts before test methods like other consts, or define inside `test_bedrock_autolog_invoke_model_embeddings` if these are specifically for the test case?

Reply:
yea we can do that, no strong preference here